calyx/src/analysis/mod.rs

```diff
@@ -6,6 +6,7 @@
 mod graph;
 mod graph_coloring;
 mod live_range_analysis;
+pub mod reaching_defns;
```

Comment:
Any reason to expose the mod instead of specific structs? In general, it'd be better to wrap the functionality in a struct and expose that instead.

Reply:
This was originally because I needed access to three things from the analysis in the pass:
1. The analysis itself
2. The constant prefix used for labeling invoke statements.
3. The GroupOrInvoke enum.

I'm pushing the invoke label stuff into utility functions associated with the analysis struct. So 2 can go. However, the unsharing pass needs to know about this enum since it has to process it as input in several places. I felt like it was bad form to include both these pieces as top level imports from the analysis crate since it breaks the association between the enum and the analysis.

Reply:
1 & 3 are good reasons I think so let's keep it.